packages/clay-drop-down/src/DropDownWithAdvancedItems.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {IItem} from './DropDownWithBasicItems';
+import Caption from './Caption';
+import ClayDropDown from './DropDown';
+import ClayDropDownGroup from './Group';
+import ClayForm from '@clayui/form';
+import Divider from './Divider';
+import Help from './Help';
+import React, {useState, useContext} from 'react';
+import Search from './Search';
+import warning from 'warning';
+
+type TType = 'checkbox' | 'group' | 'item' | 'radio' | 'radiogroup';
+
+interface IAdvancedItem extends Omit<IItem, 'type'> {
+	checked?: boolean;
+	items?: Array<IAdvancedItem>;
+	name?: string;
+	onChange?: Function;
+	separator?: boolean;
+	type?: TType;
+	value?: string;
+}
+
+interface IDropDownContentProps {
+	spritemap?: string;
+
+	/**
+	 * List of items to display in drop down menu
+	 */
+	items: Array<IAdvancedItem>;
+}
+
+interface IProps extends IDropDownContentProps {
+	action?: React.ReactElement;
```

Comment:
Since we are passing a react element here, I think its best if we label this as something like `footerContent`. `action` makes it seem like its a callback

Reply:
Yeah, I'm going with `footerContent`.